pkg/api/configmanagement/v1/types.go

```diff
@@ -569,54 +569,3 @@ type RepoList struct {
 	// Items is a list of Repo declarations.
 	Items []Repo `json:"items"`
 }
-
-// +genclient
-// +genclient:nonNamespaced
-// +kubebuilder:object:root=true
-// +kubebuilder:resource:scope=Cluster
-
-// HierarchyConfig is used for configuring the HierarchyModeType for managed resources.
-type HierarchyConfig struct {
```

Comment:
These structs still need to exist, we just don't want to generate the CRD manifests.

Reply:
Please remove the kubebuilder comments.

Reply:
Removing the kubebuilder comments would cause `make generate` to remove the `DeepCopyObject` method in zz_generated.deepcopy.go.
```
-// DeepCopyObject is an autogenerated deepcopy function, copying the receiver, creating a new runtime.Object.
-func (in *HierarchyConfig) DeepCopyObject() runtime.Object {
-       if c := in.DeepCopy(); c != nil {
-               return c
-       }
-       return nil
-}
-
```

Reply:
Is there a way to tell controller-gen to generate code but not yaml artifacts? Do we need the generated code?

Reply:
I think you can just update the make target to remove the yaml manifest after generation

Reply:
That is what I have been doing. See the changes to the `configsync-crds` Make rule.